mitiq/typing.py

```diff
@@ -125,6 +125,11 @@ class MeasurementResult:
 
     def __post_init__(self) -> None:
         # Validate arguments
+        if isinstance(self.result, dict):
```

Comment:
We could have something stricter that checks for results not being an instance of Sequence[Bitstring], which is what we expect, but then both the check and the error message become more complicated, so I would leave it as is.